src/TemplateDialog.js

```diff
@@ -0,0 +1,203 @@
+import React, { PropTypes } from 'react'
+import { connect } from 'react-redux'
+import $ from 'jquery'
+
+import LabeledSelect from './LabeledSelect'
+import LabeledTextField from './LabeledTextField'
+import DetailContainer from './DetailContainer'
+import ErrorAlert from './ErrorAlert'
+import Selectors from './selectors'
+import {
+  updateEditTemplateName,
+  updateEditTemplateDescription,
+  updateEditTemplateOS,
+  updateEditTemplateMemory,
+  updateEditTemplateCpu,
+  changeEditTemplate,
+  editTemplate,
+  closeDetail,
+} from './actions'
+
+class TemplateDialog extends React.Component {
+  constructor (props) {
+    super(props)
+    this.changeTemplate = this.changeTemplate.bind(this)
+    this.changeOperatingSystem = this.changeOperatingSystem.bind(this)
+    this.changeMemory = this.changeMemory.bind(this)
+    this.changeCpu = this.changeCpu.bind(this)
+    this.changeName = this.changeName.bind(this)
+    this.changeDescription = this.changeDescription.bind(this)
+    this.editTemplate = this.editTemplate.bind(this)
+    this.closeDialog = this.closeDialog.bind(this)
+  }
+
+  componentDidUpdate () {
+    // without this template combobox its not rerendering
+    $(this.templateOs).selectpicker('refresh')
+  }
+
+  componentDidMount () {
+    $(this.template).selectpicker()
+    $(this.templateOs).selectpicker()
+  }
+
+  changeName () {
+    this.props.changeName(this.templateName.value)
+  }
+
+  changeTemplate () {
+    this.props.changeTemplate(Selectors.getTemplateByName(this.template.value))
+  }
+
+  changeOperatingSystem () {
+    this.props.changeOperatingSystem(this.templateOs.value)
+  }
+
+  changeDescription () {
+    this.props.changeDescription(this.templateDescription.value)
+  }
+
+  changeMemory () {
+    this.props.changeMemory(this.templateMemory.value)
+  }
+
+  changeCpu () {
+    this.props.changeCpu(this.templateCpus.value)
+  }
+
+  closeDialog (e) {
+    e.preventDefault()
+    this.props.closeDialog()
+  }
+
+  editTemplate (e) {
+    e.preventDefault()
+    const template = {
+      'name': this.templateName.value,
+      'description': this.templateDescription.value,
+      'memory': this.templateMemory.value,
+      'cpu': {
+        'topology': {
+          'cores': '1',
+          'sockets': this.templateCpus.value,
+          'threads': '1',
+        },
+      },
+      'os': {
+        'type': this.templateOs.value,
+      },
+    }
+    this.props.editTemplate(template, this.props.template.get('id'))
+  }
+
+  render () {
+    return (
+      <DetailContainer>
+        <h1>Edit Template</h1>
```

Comment:
I think I will be able to wrap Operating system, memory, cpus and template to separated component.